### ISSUE TITLE
[Bug]: DecoderException while connecting to Minecraft Server play.mixelpixel.net

### ISSUE THREAD

**Issue by mfilser** [TeamJM/journeymap#825]
### What happened?

When I connect to the Minecraft Server play.mixelpixel.net the connection aborts and I get following error:
`io.netty.handler.codec.DecoderException: java.io.IOException: Packet play/clientbound/minecraft:custom_payload (class_2658) was larger than I expected, found 15 bytes extra whilst reading packet clientbound/minecraft:custom_payload`

![grafik](https://github.com/user-attachments/assets/3986de3f-4c62-4f12-800f-1bd3e614c5a5)

I am using journeymap-fabric-1.21.1-6.0.0-beta.29.jar.
After removing Journeymap from the mods folder, I can connect to the Server and play on it.

### Mod Loader

Fabric

### Minecraft Version

1.21.x

### Version

6.0.0

### Relevant log output

[latest.log](https://github.com/user-attachments/files/17691192/latest.log)
[journeymap.log](https://github.com/user-attachments/files/17691195/journeymap.log)

**Comment by mysticdrew** [comment]
Is this a server that supports multiple versions of minecraft? 

If so, are you able to connect using Minecraft and Journeymap for mc1.20.1?

**Comment by mfilser** [comment]
Unfortunately the Server only runs with Minecraft 1.21.1 to 1.21.3

**Comment by mysticdrew** [comment]
This seems like an issue with the server, with a packet they are sending. Might be hard to debug

**Comment by mfilser** [comment]
I tested it with 1.21.3, I got the same error.
Here's attached the 2 log files:
[journeymap.log](https://github.com/user-attachments/files/17692383/journeymap.log)
[latest.log](https://github.com/user-attachments/files/17692385/latest.log)

**Comment by mysticdrew** [comment]
Yes, but what about 1.21.1, since your server allows you to connect with that minecraft version?

**Comment by mfilser** [comment]
> Is this a server that supports multiple versions of minecraft?
> 
> If so, are you able to connect using Minecraft and Journeymap for mc1.20.1?

I didn't think of to write that I was playing on that Minecraft Server on version 1.20.2 and didn't have any issues, JourneyMap was running on 1.20.2 very well.
I really miss this mod a lot on 1.21.1 :wink:

**Comment by mfilser** [comment]
> Yes, but what about 1.21.1, since your server allows you to connect with that minecraft version?

This is the Minecraft Client version I have the issue I described in this Issue.
It's not running on 1.21.1

**Comment by mfilser** [comment]
I can connect to this server without JourneyMap in my mods folder, with Fabric and Vanilla Minecraft version

**Comment by mysticdrew** [comment]
> I can connect to this server without JourneyMap in my mods folder, with Fabric and Vanilla Minecraft version

What about with journeymap for 1.21.1 and minecraft 1.21.1? Please see if that works for you so I know if it is an issue with 1.21.3 or all of journeymap 6.0.0

**Comment by mfilser** [comment]
I tested this versions successfully:
- Vanilla 1.21.1
- Vanilla 1.21.3
- Fabric 1.21.1 without JourneyMap
- Fabric 1.21.3 without JourneyMap

unsuccessfully to connect:
- Fabric 1.21.1 with JourneyMap 6.0.0-beta29 works not
- Fabric 1.21.3 with JourneyMap 6.0.0-beta29 works not

Are there some version's and combinations I should test too?

**Comment by mysticdrew** [comment]
Does beta.27 work? any minecraft version.

**Comment by mfilser** [comment]
I am happy to test for you.
Here are my results:

JourneyMap is not working:
- Fabric 1.21.1, JorneyMap 6.0.0-beta27
- Fabric 1.21.1, JorneyMap 6.0.0-beta29
- Fabric 1.21.3, JorneyMap 6.0.0-beta27
- Fabric 1.21.3, JorneyMap 6.0.0-beta29

JourneyMap is working:
- Forge 1.21.1, JourneyMap 6.0.0-beta27
- Forge 1.21.1, JourneyMap 6.0.0-beta29
- Forge 1.21.3, JourneyMap 6.0.0-beta29

So I think it's an issue with Fabric 1.21.1 and 1.21.3 where JourneyMap isn't working?
As mentioned before, Fabric 1.20.2 was working (but the Server was updating to 1.21.1 about a month ago).

As I have all my other mods running on Fabric I would love to have a running version for JourneyMap too :slightly_smiling_face: 
I could also offer to help in development / debugging as I am a Java developer for more than 15 years too.

**Comment by mysticdrew** [comment]
Oh, that is interesting that forge is working just fine. Thanks for checking on that. I'll have to attempt to connect with some debugging logic to see what is going on.

**Comment by mfilser** [comment]
It was an interesting result for me too.
Your welcome, let me know if I can do anything else.
Thank you in advance for your debugging, I keep the fingers crossed you find and hopefully can change it 🙂

**Comment by mysticdrew** [comment]
is this still an issue?

**Comment by mfilser** [comment]
Hi @mysticdrew 
thank you for asking and looking into it.
In the meanwhile the server changed their code internally so the problem is solved for me.
If I am right it was something with the mod: https://www.spigotmc.org/resources/viaversion.19254/

Thank you a lot again, I appreciate your work on this great mod, especially the feature "Multi World Support" saves my gamer life a lot 🙂

**Comment by mysticdrew** [comment]
Sounds great,  thanks
